system/CompletionEvent_tests.cpp

```diff
@@ -276,6 +279,42 @@ void try_synchronizing_spawns() {
   BOOST_CHECK_EQUAL(total, N);
 }
 
+void try_iterative_spmd_gce() {
+
+  // set up re-enrolling across iterations
+  gce.enroll(cores());
+  gce.reenroll_on_complete();
+
+  static int val, stolen;
+  
+  on_all_cores( [] {
+    for(int i = 0; i < FLAGS_outer; ++i) {
+
+      if(mycore() == 0) {
+        val = 0;
+        stolen = 0;
+      }
+
+      for(int j = 0; j < FLAGS_inner; ++j) {
+        Core origin = mycore();
+        gce.enroll();
```

Comment:
Something to think about for the future, maybe we should consider having `enroll` return a "handle" which can be passed to `complete` later without worrying about what's inside. In this instance, the `CompletetionHandle` or whatever would just be a `Core`, but in the case of a `CompletionEvent.enroll` it would be a `GlobalAddress<CompletionEvent>`.

e.g.:
```cpp
auto t = gce.enroll();
send_heap_message([t]{
  gce.complete(t);
});
```

Just a thought...